### PR TITLE
test(robot): migrate test_replica_rebuild_per_volume_limit

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9774

#### What this PR does / why we need it:

migrate test_replica_rebuild_per_volume_limit

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality to delete a specified number of replicas for a volume.
	- Added a constraint ensuring only one replica can be rebuilt at a time for a volume.

- **Bug Fixes**
	- Enhanced error handling and logging for replica deletion processes.

- **Tests**
	- Added a regression test case to verify the behavior of volume replicas, ensuring only one replica is rebuilt at a time.

### CODE REVIEW

File: e2e/tests/regression/test_replica.robot [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    ...    8. Once the rebuild was completed again, verify the data checksum.
+    Given Set setting replica-soft-anti-affinity to true
+    And Create volume 0 with    numberOfReplicas=3    dataEngine=${DATA_ENGINE}
+    And Attach volume 0
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+
+    When Update volume 0 replica count to 5
+    Then Only one replica rebuilding will start at a time for volume 0
+    And Wait until volume 0 replicas rebuilding completed
```
Comment (chriscchien) [inline e2e/tests/regression/test_replica.robot:43]: Is it possible to change the keyword `Wait until volume 0 replicas rebuilding is completed` to emphasize that it ensures only one replica in the rebuild (WO) state until the rebuild is completed, for better match test case name `Test Replica Rebuilding Per Volume Limit`? Beside this, no further questions. Thank you.

File: e2e/tests/regression/test_replica.robot [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+${DATA_ENGINE}    v1
+
+*** Test Cases ***
+Test Replica Rebuilding Per Volume Limit
+    [Tags]    coretest
+    [Documentation]    Test the volume always only have one replica scheduled for rebuild
+    ...
+    ...    1. Set soft anti-affinity to `true`.
+    ...    2. Create a volume with 1 replica.
+    ...    3. Attach the volume and write a few hundreds MB data to it.
```
[the remark this answers is not in the document]
Comment (chriscchien) [reply e2e/tests/regression/test_replica.robot:29]: Agree with this comment for making test description and test code more consistent.

File: e2e/tests/regression/test_replica.robot (hunk 1 above)
Comment (yangchiu) [reply e2e/tests/regression/test_replica.robot:43]: @chriscchien Could you suggest an appropriate name?

Comment (chriscchien) [reply e2e/tests/regression/test_replica.robot:43]: How about `Only one replica for volume 0 rebuilds at a time until the rebuild is completed` for below two?
```robot
 Then Only one replica rebuilding will start at a time for volume 0
 And Wait until volume 0 replicas rebuilding completed
```

Comment (yangchiu) [reply e2e/tests/regression/test_replica.robot:44]: @chriscchien How about adding another step `Monitor only one replica rebuilding will start at a time for volume 0`:

```
Then Only one replica rebuilding will start at a time for volume 0
And Monitor only one replica rebuilding will start at a time for volume 0
And Wait until volume 0 replicas rebuilding completed
```

This aligns with the description in the test skeleton.

**Review by chriscchien — APPROVED** [review]
LGTM

**Pull request merged**